core/base_service.go

```diff
@@ -612,6 +649,39 @@ func getErrorMessage(responseMap map[string]interface{}, statusCode int) string
 	return http.StatusText(statusCode)
 }
 
+// getErrorCode: try to retrieve an error code from the decoded response body (map).
```

Comment:
I think this is more godoc compliant, but feel free to leave it as-is.
```suggestion
// getErrorCode tries to retrieve an error code from the decoded response body (map).
```